### ISSUE TITLE
elasticsearch/downloads limits tuning

### ISSUE THREAD

**Issue by mnowotka** [chembl/GLaDOS#399]
Right now elastic allows us to fetch 10k results. I believe this is a fair amount of data for rendering in a web browser. BUT!!! There are some caveats:

1. When doing substructure/similarity search this limit is only 1k because we first get ids from API and we construct a query to elastic consisting of ids we got from the API. Elastic has a limit on a query complexity but it can be changed as described here: http://george-stathis.com/2013/10/18/setting-the-booleanquery-maxclausecount-in-elasticsearch/. So I would change it from 1024 to 10k or 16384 (whatever works better) carefully checking the impact of elastic speed and mamory and CPU usage.

2. If possible when hitting a limit we shouldn't say: you got 10k results but either provide an actual number or a word 'many' not to make users think there are only 10k results in a database. Then if users adds some filters, we should add them not to 10k of selected objects but rerun query in a hope that new filters will reduce the object count below our limit.

This applies only for results presented on the web app. Downloads shouldn't have any limits and should be handled by the server side. The way I see this is actually quite complex: 

When clicking the download button for a dataset that is too large to ba handled by the client side, you put a task into query (and I guess we should use Celery for that, which nicely integrates with Django). Then you put a progress bar and fetch updates from Celery about that task. Once the task is completed we provide a download link or intiate download directly.

Since this is more complex than expected, I think the first part (1k limit -> 10k limit) should be done before SAB and the second part later in 2018.

**Comment by juanfmx2** [comment]
Please check: http://chembl-glados.herokuapp.com/substructure_search_results/c1ccccc1
Now the total count is available, the limit has been increased to 10k without having to change elastic settings, and the results are being shown while the download of the first 10k continues.

**Comment by mnowotka** [comment]
Thanks @juanfmx2 - I think this is all we can do at the moment. I'll keep that ussue open so later we can split it into smaller substasks.

**Comment by nclopezo** [comment]
This is already solved with our GLaDOS workers.
